hasktorch/src/Torch/Typed/Tensor.hs

```diff
@@ -120,12 +127,13 @@ instance (KnownNat n) => KnownDevice '( 'D.CPU, n) where
 instance (KnownNat n) => KnownDevice '( 'D.CUDA, n) where
   deviceVal = D.Device D.CUDA (natValInt16 @n)
 
-data Tensor (device :: (D.DeviceType, Nat)) (dtype :: D.DType) (shape :: [Nat]) where
+
+data Tensor (device :: (D.DeviceType, Nat)) (dtype :: D.DType) (shape :: Shape) where
   UnsafeMkTensor :: forall device dtype shape. {toDynamic :: D.Tensor} -> Tensor device dtype shape
 
```

Comment:
This pr replaces  tensor's shape( [Nat] ) with  [Type->Type].
The codes become harder to understand the code.
I think the boundary between Named Tensor and non-Named Tensor is vague.